ed_sensor_integration/tools/segmenter.cpp

```diff
@@ -264,79 +273,260 @@ int main(int argc, char **argv)
-                    bb_max.y = std::max(bb_max.y, y);
+        // needed parameters: Fitterdata data;
+        int sensor_x = canvas_width/2;
+        int sensor_y = canvas_height * 8/10;
+        float canvas_resolution = 100; //pixels per meter
 
-                    for(double x2 = f * x; x2 < (f * (x + 1)); ++x2)
-                        for(double y2 = f * y; y2 < (f * (y + 1)); ++y2)
-                            canvas.at<cv::Vec3b>(y2, x2) = cv::Vec3b(0, 0, 255);
-                }
+        cv::Point sensorlocation(sensor_x, sensor_y);
+        cv::Scalar sensorcolor(0,255,0);
+        cv::circle(canvas, sensorlocation, 3, sensorcolor, CV_FILLED);
 
-                cv::Point d(2, 2);
-                cv::rectangle(canvas, f * bb_min, f * bb_max, cv::Scalar(255, 255, 255), 2);
-                cv::rectangle(canvas, f * bb_min - d, f * bb_max + d, cv::Scalar(0, 0, 0), 2);
-                cv::rectangle(canvas, f * bb_min + d, f * bb_max - d, cv::Scalar(0, 0, 0), 2);
-            }
+        // paint sensor_ranges
+        for(unsigned int i = 0; i < data.sensor_ranges.size(); ++i){
+            float a = (((float)i - 100.0)/100.0);// TODO remove hardcoded values: add this info to fitterdata
+            float x_m = data.sensor_ranges[i] * sin(a);
+            float y_m = data.sensor_ranges[i] * cos(a);
 
-        }
-        else
-        {
-            std::cerr << error_msg << std::endl;
-            cv::putText(canvas, "Segmentation failed", cv::Point(10, 20), cv::FONT_HERSHEY_COMPLEX_SMALL, 1, cv::Scalar(255, 255, 255), 1);
-        }
+            // postion to pixels
+            int x_p = sensor_x + (int)(x_m * canvas_resolution);
+            int y_p = sensor_y - (int)(y_m * canvas_resolution);
 
-        cv::imshow("RGB", canvas);
-        char key = cv::waitKey();
+            if (x_p < 0 || x_p >= canvas_width)
+                continue;
+            if (y_p < 0 || y_p >= canvas_height)
+                continue;
+            if (data.sensor_ranges[i] == 0) // filter out sensor_ranges equal to zero
+                continue;
 
-        if (key == 81)  // Left arrow
-        {
-            if (i_snapshot > 0)
-                --i_snapshot;
+            // paint to screen
+            cv::Point centerCircle(x_p, y_p);
+            cv::Scalar colorCircle(0,0,255);
+            cv::circle(canvas, centerCircle, 2, colorCircle, CV_FILLED);
         }
-        else if (key == 83) // Right arrow
-        {
-            ++i_snapshot;
+
+        //std::cout << "range: ("<< data.sensor_ranges.size() << ")" << std::endl;
+
+    // paint entity (from worldmodel)
+        EntityRepresentation2D entity_2d = fitter.GetOrCreateEntity2D(e);
+
+        geo::Pose3D pose = e->pose();
+        //float yaw_ent = atan2(pose.R.yx, pose.R.xx)
+
+        // entity yaw
+        float yaw_ent = atan2(pose.R.yx, pose.R.xx);
+
+        // robot yaw
+        float yaw_robot = atan2(snapshot.sensor_pose.R.yx, snapshot.sensor_pose.R.xx);
```

Comment:
The reason this simple formula works for the yaw of the entity is that there is only rotation around the gravity axis. The sensor pose is full rotation (x,y,z). Getting its yaw pose is slightly more complex